### ISSUE TITLE
Update the modal / dialog component to refer the current release of A11yDialog

### ISSUE THREAD

**Issue by joesnellpdx** [10up/component-library#44]
**Is your feature request related to a problem? Please describe.**
[A11yDialog](https://github.com/KittyGiraudel/a11y-dialog) made breaking changes in its 7.0 release. Our documentation refers to a legacy version of A11yDialog and we should update our documentation to refer to the latest release. 

**Describe the solution you'd like**
We want to rewrite our docs using the new version (Version 7) and have a notice in there that Version 6 has been deprecated with links (see blow).
In other words, replace all the current documentation to follow version 7's usage / etc and a warning that version 6 will require a migration to work with the recent update.

- [Links from Kitty on twitter](https://twitter.com/KittyGiraudel/status/1374360536574464007)
- [Current docs](https://a11y-dialog.netlify.app/)
- [Baseline docs](https://baseline.10up.com/component/modal/)

**Describe alternatives you've considered**
N/A
